src/sidebar/service-url.js

```diff
@@ -3,44 +3,91 @@
 var urlUtil = require('./util/url-util');
 
 /**
- * A map of all route names to relative URLs on the Hypothesis service that
- * the client links to.
+ * An AngularJS service that returns an absolute URL given a link name and
```

Comment:
The problem is that if I change this docstring to document the outer function: "A factory that returns a function that..." and then document its `annotationUI` and `store` params, then _that_ outer docstring (which isn't very useful) will be the one that appears in the docs generated by jsdoc. Even if I do put a docstring on the inner function (and there document the actual `linkName` and `params` params) jsdoc does not extract that docstring.

It _would_ work if the inner function was actually defined as `function serviceUrl() {...}` at top-level, then the outer function could be renamed `function serviceUrlFactory()` and could still `return` the inner function, and both `serviceUrlFactory` and `serviceUrl` would appear in the jsdoc-generated HTML. Unfortuntely the inner function is a closure...

So the awkwardness remains that the docstring is on the factory function, but actually documents the params of the inner function. The rest of the docstring therefore needs to be a docstring for the inner function as well (so not "A factory that returns...").

Or I change the docstring to document the factory function and its params (`annotationUI` and `store`) but also, in the same docstring, document the params of the function that the factory function returns? But I don't think this is the semantics that jsdoc supports.

Anyway, I've left it as a docstring that documents the inner function, not the outer one, but I've changed it to "A function that..." instead of "An AngularJS service..."